rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,358 @@
+endpoints and how they are used. These are the new API endpoints needed to complete an upload.
+
+These APIs replace the [PUT /v1/submission/{project_uuid}/file](https://docs.google.com/document/d/1d8tv2Ub5b3E7Il85adOAUcG8P05N6UBZJ3XbhJSRrFs/edit#heading=h.3hln6w2kzoyt)
+section.
+
+#### PUT submission/{submission_id}/upload/link
+
+An authenticated user can upload a file from a shared link to a data set in their submission.
+
+If the upload is in an error state, this endpoint can be used with the dataset_id to restart the submission. If a new
+link is provided, the new link will be used. If the submissions are not in an error state or complete, this endpoint
+will return the current status of the submission.
+
+Starting an upload causes the database to be updated with a new data set. The dataset will set the initial state of the upload
+to waiting once it has been accepted and is in the download queue.
+
+**Request:**
+
+| Parameter             | Description                                                                                                                                           |
+| --------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Link                  | a shared link to the file                                                                                                                             |
+| submission_id         | identifies the submission                                                                                                                             |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used to check the status of the upload.                                                                        |
+| _Optional_ overwrite  | A boolean, indicating if the user would like to overwrite the file that was uploaded. The upload status must be complete for the dataset_id provided. |
+
+**Response:**
+
+| Key        | Description                                                                |
+| ---------- | -------------------------------------------------------------------------- |
+| dataset_id | identifies the data set the file will be associated with after validation. |
+| status     | Provides the current status of the upload                                  |
+
+**Error Responses:**
+
+| Code | Description                                                                                                      |
+| ---- | ---------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress. |
+| 400  | if the file type is invalid                                                                                      |
+
+#### Delete submission/{submission_id}/upload
```

Comment:
Does this API work for both cancelling a shared link upload and browser upload, or just shared link upload?

If it's just for shared link upload, then should we make it consistent and use:

`DELETE submission/{submission_id}/upload/link`?

Reply:
It was the plan to use it for both browser and cloud. What is your opinion on doing this?

Reply:
Now knowing the plan for `PUT` above, having one API for both link and browser makes sense! Thank you!